src/js/control-bar/progress-control/mouse-time-display.js

```diff
@@ -53,11 +67,40 @@ class MouseTimeDisplay extends Component {
 
     this.el().style.left = position + 'px';
     this.el().setAttribute('data-current-time', time);
+
+    if (this.keepTooltipsInside) {
+      let clampedPosition = this.clampPosition(position);
+      let difference = position - clampedPosition + 1;
+      let tooltipWidth = parseFloat(window.getComputedStyle(this.tooltip).width);
+      let tooltipWidthHalf = tooltipWidth / 2;
+
+      this.tooltip.innerHTML = time;
+      this.tooltip.style.right = `-${tooltipWidthHalf - difference}px`;
+    }
   }
 
   calculateDistance(event) {
     return Dom.getPointerPosition(this.el().parentNode, event).x;
   }
+
+  clampPosition(position) {
```

Comment:
Seems like this might deserve a doc block to explain what "clamping" position is.